frontend/src/components/dirent-list-view/dirent-list-item.js

```diff
@@ -412,6 +412,19 @@ class DirentListItem extends React.Component {
     if (currentRepoInfo.permission === 'cloud-edit' || currentRepoInfo.permission === 'preview') {
       return '';
     }
+
+    let isShowShareBtn = false;
+    if (this.props.showShareBtn) {
+      if (canGenerateShareLink) {
+        isShowShareBtn = true;
+      } else {
+        if (dirent.type === 'dir') {
+          isShowShareBtn = true;
+        } else {
+          isShowShareBtn = false;
+        }
+      }
+    }
```

Comment:
let isShowShareBtn = (dirent.type === 'dir' && this.props.isShowShareBtn) || canGenerateShareLink;